internal/scheduler/scheduler.go

```diff
@@ -129,19 +134,41 @@ func worker(ctx context.Context, workerID int, crashTrackerClient crashtracker.C
 	for {
 		select {
 		case job := <-jobQueue:
-			log.Ctx(ctx).Debugf("Worker %d processing job: %s", workerID, job.GetName())
-			if err := job.Execute(ctx); err != nil {
-				msg := fmt.Sprintf("error processing job %s on worker %d", job.GetName(), workerID)
-				// call crash tracker client to log and report error
-				crashTrackerClient.LogAndReportErrors(ctx, err, msg)
-			}
+			executeJob(ctx, job, workerID, crashTrackerClient, tenantManager)
 		case <-ctx.Done():
-			log.Ctx(ctx).Infof("Worker %d stopping...", workerID)
+			log.Infof("Worker %d stopping...", workerID)
 			return
 		}
 	}
 }
 
+// executeJob executes a job and reports any errors to the crash tracker.
+func executeJob(ctx context.Context, job jobs.Job, workerID int, crashTrackerClient crashtracker.CrashTrackerClient, tenantManager tenant.ManagerInterface) {
+	// Handle multi-tenant jobs.
+	if job.IsJobMultiTenant() {
+		tenants, err := tenantManager.GetAllTenants(ctx)
+		if err != nil {
+			msg := fmt.Sprintf("error getting all tenants for job %s on worker %d", job.GetName(), workerID)
+			crashTrackerClient.LogAndReportErrors(ctx, err, msg)
+			return
+		}
+		for _, t := range tenants {
+			log.Ctx(ctx).Debugf("Processing job %s for tenant %s on worker %d", job.GetName(), t.ID, workerID)
+			tenantCtx := tenant.SaveTenantInContext(ctx, &t)
```

Comment:
What do you think about creating a new context here instead of using the "same" context to save the tenants?

We avoid saving a tenant in a context that belongs to another tenant.
```suggestion
			tenantCtx := tenant.SaveTenantInContext(context.Background(), &t)
```

Reply:
addressed in 2fa448844c0770c1c436647595e173fa5f2ca762